lasy/utils/denoise.py

```diff
@@ -0,0 +1,103 @@
+from lasy.profiles.transverse.hermite_gaussian_profile import (
+    HermiteGaussianTransverseProfile,
+)
+from lasy.profiles.transverse.transverse_profile import TransverseProfile
+from lasy.utils.mode_decomposition import hermite_gauss_decomposition
+
+
+def denoise_transverse_hg(
+    transverse_profile,
+    wavelength,
+    resolution=0.2e-6,
+    lo=[-2e-4, -2e-4],
+    hi=[2e-4, 2e-4],
+    n_modes_x=2,
+    n_modes_y=2,
+):
+    """
+    Denoise the transverse profile by decomposing it into a set of Hermite-Gaussian modes.
+
+    The profiles are weighted according to mode coefficients and then added.
+
+    Parameters
+    ----------
+    transverse_profile : class instance
+        An instance of a class or sub-class of TransverseProfile. Or it could be
+        a path to an image file from which intensity data will be read.
```

Comment:
Let avoid ambiguity in the input parameter: it has one type, simple. In order to read an image, the user would create their own `TransverseProfileFromData` and pass it there ok?
```suggestion
        An instance of a class or sub-class of TransverseProfile.
```